src/main/java/com/julienvey/trello/domain/TrelloEntity.java

```diff
@@ -10,7 +10,16 @@
     @JsonIgnore
     protected Trello trelloService;
 
-    public void setInternalTrello(Trello trelloService) {
+    @SuppressWarnings("unchecked")
+    public <T extends TrelloEntity> T setInternalTrello(Trello trelloService) {
         this.trelloService = trelloService;
+        return (T) this;
+    }
+
+    protected Trello getTrelloService() {
+        if (trelloService == null)
+            throw new IllegalStateException("The trelloService not initialized. Please call setInternalTrello before.");
+
+        return trelloService;
```

Comment:
if the goal it to return the value previously set then `setInternalTrello` and `getTrelloService` looks inconsistent.
while `setInternalTrello` is a weird legacy name, the corresponding getter should be consistent with it.
we can rename them both in a separate mass-refactoring commit.

about null check and exception - let's  skip that part. while I like the idea of proving a meaningful exception, it is not up to the getter method to decide whether or not an empty value constitutes an error in all situations. 
consuming code should decide if that is an acceptable value.

Reply:
The goal is keep users away from using direct access to the field.

If we rename something like you mentioned already we can break users code.

It's a not and data class getter method. TrelloEntity has a state and IMO there is no situation (in subclasses) where trelloSevice is needed and it is null.